remote/remote.go

```diff
@@ -156,14 +157,21 @@ func (r *Remote) ResolveHeadRef(ctx context.Context, peername, name string) (*re
 	return ref, err
 }
 
-// RemoveDatasetRef handles requests to remove a dataset
-func (r *Remote) RemoveDatasetRef(ctx context.Context, params map[string]string) error {
+// RemoveDataset handles requests to remove a dataset
+func (r *Remote) RemoveDataset(ctx context.Context, params map[string]string) error {
```

Comment:
This shift to removing both persisted versions and a ref kinda breaks the original intention of the `/refs` endpoint of the remote api. It's a little easier to see from the api routes definition: https://github.com/qri-io/qri/blob/902927181e03c9e68510528d85e842b6b0b91d05/api/api.go#L248-L255

The purpose of the `/refs` endpoint predates our logbook work, used to both resolve alias references with HEAD paths, and remove a ref on an unpublish request. There's some work sewing in support for deleting single dags in dsync.

That being said, there's nothing in dsync that supports the idea of removing multiple generations of a history at once, which we need _asap_ So let's merge this & start refactoring past it.

We have a few options for long-term solutions:
1. Multi-rooted DAG manifests, and have `dsync` interpret remove requests as a request to remove all root hashes recursively.
2. Create a higher-order route for deleting logs, references, and persisted versions in a single endpoint. We can use logsets in that context to determine which dags to remove.